functorch/functorch/_src/aot_autograd.py

```diff
@@ -256,6 +281,12 @@ def fake_fn(primals, tangents):
                     compiled_bw = bw_compiler(bw_module, bw_args)
             else:
                 fw_outs = normalize_as_list(compiled_fw(*flat_tensor_args))
+                if config.use_functionalize:
```

Comment:
I think we're going to need to pass this input mutation submodule to backends as well, potentially.

Reply:
hmm. What do you think is the benefit of sending it to backends - do you think there's potential for a backend to optimize those `copy_()` in some ways?

My mental model was basically:
(1) Create a giant purely functional graph, let the backend go wild and optimize it
(2) If there are any input mutations afterwards, just run the eager code to patch up the inputs (I assumed this isn't worth optimizing, but maybe that's wrong)